terraform/modules/bb_dashboard/main.tf

```diff
@@ -0,0 +1,731 @@
+##
+#
+# NOTE: This module is for defining a main CloudWatch dashboard
+#
+##
+
+resource "aws_cloudwatch_dashboard" "main-dash" {
+   count          = "${var.dashboard_enable}"
+
+   dashboard_name = "MainDashboard-${var.app}-${var.env}"
+   dashboard_body = <<EOF
```

Comment:
For large blocks of embedded json, I sometimes find it helpful to store the contents in a separate file and use terraform's [template_file](https://www.terraform.io/docs/providers/template/d/file.html) data source to include in the `.tf` file.